pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -621,3 +621,36 @@ func validateTaskArraysIsolated(value, prefix string, arrayNames sets.String) *a
 func isParamRefs(s string) bool {
 	return strings.HasPrefix(s, "$("+ParamsPrefix)
 }
+
+// ValidateParamArrayIndex validates if the param reference to an array param is out of bound.
+// error is returned when the array indexing reference is out of bound of the array param
+// e.g. if a param reference of $(params.array-param[2]) and the array param is of length 2.
+// - `trParams` are params from taskrun.
+// - `taskSpec` contains params declarations.
+func (taskSpec *TaskSpec) ValidateParamArrayIndex(ctx context.Context, params []Param) error {
```

Comment:
same as pipeline_validation.go, the linting is failing with:

```
pkg/apis/pipeline/v1beta1/task_validation.go:630:1: receiver-naming: receiver name taskSpec should be consistent with previous receiver name ts for TaskSpec (revive)
```

Please change this function declaration to:

```
func (ts *TaskSpec) ValidateParamArrayIndex(ctx context.Context, params []Param) error {
```

https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/task_validation.go#L70